Add a new blog and a blog post (#466)
haiku/website: content/blog/jt15s/2020-12-31_hello_from_the_haiku_promotion_team.md

+++
type = "blog"
title = "Hello from the Haiku Promotion 'Team'!"
author = "jt15s"
date = "2020-12-31 21:33:25+11:00"
tags = ["haiku", "software"]
+++
Hi there, I'm jt15s, a new Haiku community member! I've been following the project for a few years now since I stumbled upon an article about Haiku. 

> If you read the title (which I'm assuming you did), you’re probably going, "wait, we have a promotion team now?" Well, yes, now we do. As an informal “team” of two (currently), we are super excited to share our plans to help promote Haiku and make more people aware of it.

Author: Thanks for the info! I will add that in.

> ## Who's in it?

Author: @humdingerb I've changed it to "Who's in the team".

> Me, as well as scott_puopolo18 as of now. We are also working with Haikunauts and BeSly, two Haiku-related resources. However, anyone who wishes to help is more than welcome to! Reach out to me (jt15s) [on the forums](https://discuss.haiku-os.org) if you are interested. If you don't have a forum account yet, create one - a forum account is free! We are also looking for the following informal roles:

Author: @humdingerb I've fixed those parts and have added my real name. I'm just waiting for Scott's reply now.

> Me, as well as scott_puopolo18 as of now. We are also working with Haikunauts and BeSly, two Haiku-related resources. However, anyone who wishes to help is more than welcome to! Reach out to me (jt15s) [on the forums](https://discuss.haiku-os.org) if you are interested. If you don't have a forum account yet, create one - a forum account is free! We are also looking for the following informal roles:

Reviewer: > Using nicknames is fine too. I like to normalize that, so that people not willing to reveal their real name for whatever reason don't feel singled out.

I suppport that in general; I'm pretty much only known by my nickname.
In case of the PR&marketing team though, I can imagine that their contact to business types and journalists may be easier with the additional establishment of trust by knowing the real name of your counterpart.
Don't feel obliged to, though.

> Me, as well as scott_puopolo18 as of now. We are also working with Haikunauts and BeSly, two Haiku-related resources. However, anyone who wishes to help is more than welcome to! Reach out to me (jt15s) [on the forums](https://discuss.haiku-os.org) if you are interested. If you don't have a forum account yet, create one - a forum account is free! We are also looking for the following informal roles:

Author: @pulkomandy You're right, it is better to do so in English. I will tweak that.

> Me, as well as scott_puopolo18 as of now. We are also working with Haikunauts and BeSly, two Haiku-related resources. However, anyone who wishes to help is more than welcome to! Reach out to me (jt15s) [on the forums](https://discuss.haiku-os.org) if you are interested. If you don't have a forum account yet, create one - a forum account is free! We are also looking for the following informal roles:

Author: @pulkomandy I'll create a ticket asking for the list to be updated, but obviously, I'll have to wait for a little while to see if others want to join the team.

> * Video Editor - it would be good to have someone produce video content for Haiku. Videos wouldn’t need to be too complex: just something that looks simple yet professional and something less than 10 minutes long.

Author: Done!

* Presenter - we need someone who is a confident speaker and who can speak in front of a camera (preferably with a good camera setup too) to be the face (or voice) of any Haiku videos we make, to give them a more “human touch”.

> * Graphic Designer - it would be nice if we had someone who was solely responsible for producing graphics for Haiku promotion. Graphics should be uploaded to both the website and Haiku Artwork repository using Github, but I’m more than happy to help teach people how to do that.

Author: Good point. I've edited that line to reflect your suggestion.

* Live Stream Technician - we are considering doing live streams, but we will need someone with technical knowledge of live-streaming who can help facilitate this.

> * Events Coordinator - it would be good to have someone search for events which Haiku could take part in (i.e. FOSS conferences) and help organise Haiku’s participation in these events.

Author: Noted. I've deleted the text about Events.

## Our promotional strategy

> As the promotion team, we want to make sure people know about Haiku. One of the biggest problems facing the Project is that we are more obscure than other FOSS projects. This means it's harder to get new contributors, whether this is developers, documentation writers or just volunteers in general. Hence we aim to attract more attention towards Haiku, and in turn, new users and volunteers. 

Reviewer: ```suggestion
As the promotion team, we want to make sure people know about Haiku. One of the biggest problems facing the project is that we are more obscure than other FOSS projects. This means it's harder to get new contributors, whether this is developers, documentation writers or just volunteers in general. Hence we aim to attract more attention towards Haiku, and in turn, new users and volunteers. 
```

> As the promotion team, we want to make sure people know about Haiku. One of the biggest problems facing the project is that we are more obscure than other FOSS projects. This means it's harder to get new contributors, whether this is developers, documentation writers or just volunteers in general. Hence we aim to attract more attention towards Haiku, and in turn, new users and volunteers. 

Author: I feel like Haiku is in the more obscure category: I feel if I asked any random stranger on the street they would have not heard of Haiku - maybe one would remember it from its earlier days but I wouldn't count on it. The chances of a younger person having heard of Haiku are even less - most of them might not even know about BeOS. Hence why we need more awareness and attention, such as that article on Lilliputing that mentioned how Medo fits on a floppy disk (of all the headlines they could have made, they used that one...).

> As the promotion team, we want to make sure people know about Haiku. One of the biggest problems facing the project is that we are more obscure than other FOSS projects. This means it's harder to get new contributors, whether this is developers, documentation writers or just volunteers in general. Hence we aim to attract more attention towards Haiku, and in turn, new users and volunteers. 

Reviewer: Since we cannot take an accurate survey of just how obscure Haiku is, while painting ourselves as a bit of an underdog is sensible from a PR point of view, you could just try making it less strong, e.g.:

```suggestion
As the promotion team, we want to make sure people know about Haiku. One problem facing the project is that we may be more obscure than other FOSS projects. This would mean it's harder to get new contributors, whether this is developers, documentation writers or just volunteers in general. Hence we aim to attract more attention towards Haiku, and in turn, new users and volunteers. 
```

How will we do this, you might ask? Well, there are a couple of aspects of our strategy. 

### The media and reviewers
A database of all the reviews/mentions of Haiku is being compiled for easy reference and so we can contact a variety of media sources when we have a new release. Additionally, we are looking for more technology-related channels who may have not mentioned or reviewed Haiku before and see if they are interested in having a closer look at the OS. 

In terms of the media in general, we are planning to prepare press releases for the media - a press release is simply a formal, properly formatted version of an announcement. By releasing press releases it makes it easier for other media organisations to find major Haiku announcements and run a story about Haiku. 

In terms of reviewers, as mentioned before we are planning to contact reviewers and media organisations once Beta 3 is released. We have also prepared a reviewer information sheet with tips for reviewers, which is in the draft stage. 

> ### Businesses

Author: @pulkomandy Thanks! I've added that in and will contact mmu_man to see if I can get the poster and see if he's interested in helping us.

Businesses can benefit from Haiku as an operating system, and personally, I see Haiku as a lightweight alternative to the proprietary desktop operating systems businesses currently use. 

An information sheet for businesses has been drafted, which gives businesses a rundown of Haiku and its advantages and disadvantages as a business-oriented operating system. However, to attract more businesses we must improve Haiku further, and hence we will delay marketing efforts for businesses until Haiku reaches a stable and mature stage. 

> 

Reviewer 2: It's a bit strange to me that this strategy mentions developers and businesses, but skips over the main focus of Haiku (in it's finalized form): regular users, running them at home on their personal computers.

While we're not quite ready for a general release (my personal metric is "would I install Haiku on my parent's computer?"), we already have some early adopters users. People who appreciate Haiku for what it is and what it tries to be, and are ready to live with the limitations. This is also something we can build upon, even if it needs some care to not set too high expectations.

Author: @pulkomandy Agreed - thanks for pointing that out! I will write a paragraph mentioning general users as soon as I can.

### Partnerships with other Haiku-related sources

> If you haven't heard of them already, BeSly and Haikunauts are the two major third party Haiku sources. BeSly is a wiki with tips and tricks on using Haiku, whilst Haikunauts is a blog aimed at promoting Haiku as an OS. As mentioned above, we will work in partnership with these two sources and see what we can do together to help spread the word about Haiku.

Reviewer: Still think links would be appropriate...

> If you haven't heard of them already, BeSly and Haikunauts are the two major third party Haiku sources. BeSly is a wiki with tips and tricks on using Haiku, whilst Haikunauts is a blog aimed at promoting Haiku as an OS. As mentioned above, we will work in partnership with these two sources and see what we can do together to help spread the word about Haiku.

Author: Thanks for suggesting that - great idea. Do you know how I can contact DaaT? His forum profile hasn't been active since 2016.

Additionally, we are also eager to work with any smaller blogs, websites or channels. If you run a blog, website, channel, or any other source, please contact me via the forums and let me know!

### General
I am thinking of running an advertisement in Read the Docs, a documentation website that serves a wide array of open-source projects. Read the Docs allocates a portion of their advertising inventory to FOSS projects such as ourselves, hence this is a good opportunity to promote Haiku. What will go on the advertisement has not been decided as of now.

scott_puopolo18 is working on a set of interview questions for our developers. By interviewing the developers we can give the public a glimpse into the world of Haiku development, as well as shining a light on the challenges and opportunities in an open-source project. The predetermined set of questions means that we don't have to write new questions for each developer: to save time, we can simply send them off and the developer can write their response.

> We are also planning to distribute more content via our often-underutilised social media platforms, as well as update them with current information (i.e. adding profile page art, applying for verification, updating or adding links etc.). Short-form posts on social media are what immediately comes to mind, although we are considering live-streams or videos if there is someone willing to help out.

Reviewer: ```suggestion
We are also planning to distribute more content via our often under-utilised social media platforms, as well as update them with current information (i.e. adding profile page art, applying for verification, updating or adding links etc.). Short-form posts on social media are what immediately comes to mind, although we are considering live-streams or videos if there is someone willing to help out.
```

Haikunauts also has several other ideas for promotional events, but we will share these with you later as we are still working out the details of these events.

> ### Closing Words

Reviewer: ```suggestion
### In closing
```

> We are super excited to see how we can promote Haiku and help the project gain new members and volunteers. **Remember, if you’d like to join, don’t hesitate in sending me a private message on the forums!**

Reviewer: ```suggestion
We are super excited to see how we can promote Haiku and help the project gain new members and volunteers. **Remember, if you’d like to join, don’t hesitate and send me a private message on the forums!**
```

How will we do this, you might ask? Well, there are a couple of aspects of our strategy. 

### The media and reviewers
A database of all the reviews/mentions of Haiku is being compiled for easy reference and so we can contact a variety of media sources when we have a new release. Additionally, we are looking for more technology-related channels who may have not mentioned or reviewed Haiku before and see if they are interested in having a closer look at the OS. 

In terms of the media in general, we are planning to prepare press releases for the media - a press release is simply a formal, properly formatted version of an announcement. By releasing press releases it makes it easier for other media organisations to find major Haiku announcements and run a story about Haiku. 

In terms of reviewers, as mentioned before we are planning to contact reviewers and media organisations once Beta 3 is released. We have also prepared a reviewer information sheet with tips for reviewers, which is in the draft stage. 

### General Users

> We musn't forget about what is probably our main user group: general users. General users of Haiku are those who come to explore and try out the OS. Due to Haiku's slightly-buggy nature (we are in Beta after all), our general users are typically more understanding of the limitations and issues with the project. Attracting more general users and making sure our general userbase is content is very important.

Reviewer: ```suggestion
We mustn't forget about what is probably our main user group: general users. General users of Haiku are those who come to explore and try out the OS. Due to Haiku's remaining rough edges and still lurking bugs (we are in Beta after all), our general users are typically more understanding of the limitations and issues with the project. Attracting more general users and making sure our general userbase is content is very important.
```

Additionally, we want to make any existing ports of FOSS projects to Haiku official platforms - this means having Haiku listed as an officially supported platform on the project's website. By getting Haiku listed on projects such as KDE, more people will be aware of Haiku and the fact that Haiku has a multitude of apps available to download.

### Researchers

> Surprisingly, Haiku is also an excellent system for research purposes. Let me explain further. Researchers don't need the stability and reliability that businesses do, which means Haiku is a possible research system for them. Furthermore, it is possible to make invasive and diruptive changes to Haiku without having to dodge security restrictions imposed by the system, or having to "hack" their way into the system. In the past, Haiku has had successful collaboration with Auckland University, where a group working on UI/UX research experimented with Haiku. 

Reviewer: ```suggestion
Surprisingly, Haiku is also an excellent system for research purposes. Let me explain further. Researchers don't need the stability and reliability that businesses do, which means Haiku is a possible research system for them. Furthermore, it is possible to make invasive and disruptive changes to Haiku without having to dodge security restrictions imposed by the system, or having to "hack" their way into the system. In the past, Haiku has had successful collaborations with the Auckland University, where a group working on UI/UX research experimented with Haiku. 
```

> They produced Stack&Tile as well as the Auckland Layout for laying out user interfaces using a constraint solver. The layout system in Apple's iOS today is based on this research work. Apart from the Auckland Layout, the group also had other projects which were not implemented into Haiku, such as a tool to auto-generate user documentation from user interface code as well as a way to dynamically modify the layout of a running window to optimise it for a specific workflow.

Reviewer: ```suggestion
They produced Stack&Tile as well as the Auckland Layout Engine (ALE) for laying out user interfaces using a constraint solver. The layout system in Apple's iOS today is based on this research work. Apart from ALE, the group also had other projects which were not implemented into Haiku, such as a tool to auto-generate user documentation from user interface code as well as a way to dynamically modify the layout of a running window to optimise it for a specific workflow.
```

If you follow technology news, you might have seen a mention of Medo, the recently-released 4K editor clocking in at only 1.44MB! This number alone highlights one of Haiku's many development advantages: our apps are small and light - even if you have a project with lots of features crammed into it, on Haiku the file size will likely be very reasonable. Add that to the fact that we have HaikuPorts, an organised porting system, that we use C++, one of the most popular progamming languages in the world, and that we are compatible with frameworks like Qt, a popular UI frontend framework, and you get the development environment that is Haiku. We want to attract more app developers, whether that's developers who want to port their existing app over to Haiku, or new developers who wish to make an app in Haiku that they feel is needed.
Additionally, we want to make any existing ports of FOSS projects to Haiku official platforms - this means having Haiku listed as an officially supported platform on the project's website. By getting Haiku listed on projects such as KDE, more people will be aware of Haiku and the fact that Haiku has a multitude of apps available to download.

> ### Researchers

Reviewer 2: I found mmu_man's poster about this: http://eurosys2010.sigops-france.fr/poster_demo/eurosys2010-final17.pdf

It also mentions Axel's work on the filesystem layer and Colin's work on Wifi (https://www.haiku-os.org/news/2016-02-26_wlan_master_thesis_published/) support were both done as master thesis in universities. I think this was also the case for Rudolf's NVidia driver: https://www.haiku-os.org/legacy-docs/writing-video-card-drivers/

And, maybe if we extend this to "research and education", we can also talk about GSoC and http://semesterofcode.com (we "participated" in the latter but received no student applications)

Author: @pulkomandy Do you have a link to Axel's work on the filesystem layer?

Reviewer 2: No, I couldn't find it.

Surprisingly, Haiku is also an excellent system for research purposes. Let me explain further. Researchers don't need the stability and reliability that businesses do, which means Haiku is a possible research system for them. Furthermore, it is possible to make invasive and disruptive changes to Haiku without having to dodge security restrictions imposed by the system, or having to "hack" their way into the system. In the past, Haiku has had successful collaborations with the Auckland University, where a group working on UI/UX research experimented with Haiku. 

> They produced Stack&Tile as well as the Auckland Layout Engine (ALE) for laying out user interfaces using a constraint solver. The layout system in Apple's iOS today is based on this research work. Apart from ALE, the group also had other projects which were not implemented into Haiku, such as a tool to auto-generate user documentation from user interface code as well as a way to dynamically modify the layout of a running window to optimise it for a specific workflow.

Reviewer 3: It would be nice to have links here that prove claims such as that Apple's iOS layout system is based on work done by the Auckland University on Haiku.

Reviewer 2: The documentation from Apple: https://developer.apple.com/library/archive/documentation/UserExperience/Conceptual/AutolayoutPG/index.html

I don't think it credits the Auckland University explicitly, but the concepts are very similar.

Actually it predates the Auckland Layout Editor (Apple had this since 2011 and the paper from Auckland university was published in 2013: https://www.researchgate.net/publication/262402078_The_Auckland_Layout_Editor_An_improved_GUI_layout_specification_process)

I'm not sure how later versions of the layout system at Apple got ideas from that paper, as they kept adding things in later versions of their system (for example a summary of their changes is here: https://sarunw.com/posts/history-of-auto-layout-constraints/ )

Author: Since we don't know if Apple based their system on the ALE or not, I've just removed the line saying that. I've also added a link to the research paper too.

> They produced Stack&Tile as well as the [Auckland Layout Engine (ALE)](https://www.researchgate.net/publication/262402078_The_Auckland_Layout_Editor_An_improved_GUI_layout_specification_process) for laying out user interfaces using a constraint solver. Apart from ALE, the group also had other projects which were not implemented into Haiku, such as a tool to auto-generate user documentation from user interface code as well as a way to dynamically modify the layout of a running window to optimise it for a specific workflow.

Reviewer 2: > the group also had other projects which were not implemented into Haiku

This wording confuses me a bit. These things were implemented into their testing version of Haiku, but not in the official version. The way you wrote it it sounds a bit like they used a different opeating system for these other projects.

Also, maybe this page is a better entry point instead of linking directly to the research paper for ALE: https://www.cs.auckland.ac.nz/~lutteroth/research.html
There are some videos there, as well as a maybe more easily accessible description of the work.

> Apart from Auckland University, mmu_man, a member of the Haiku Community, also published a research paper about the interoperability of filesystem attributes.

Reviewer 2: The paper about filesystem attributes: https://dcpapers.dublincore.org/pubs/article/view/3633/1859

Also an article about using QEMU for operating system development at the QEMU user forum: http://adt.cs.upb.de/quf/quf11/QUF11-papers/quf2011-11.pdf

> We aim to try to attract more research groups by promoting the uses of Haiku for research, and we also want to try to reach out to Auckland University and see if their UI research group is interested in coming back to Haiku to conduct any future research.

Reviewer: ```suggestion
We aim to attract more research groups by promoting the uses of Haiku for research, and we also want to reach out to Auckland University and see if their UI research group is interested in coming back to Haiku to conduct any future research.
```
As Master Yoda said: "Do or do not. There is no try." :)

### Businesses
Businesses can benefit from Haiku as an operating system, and personally, I see Haiku as a lightweight alternative to the proprietary desktop operating systems businesses currently use. 

An information sheet for businesses has been drafted, which gives businesses a rundown of Haiku and its advantages and disadvantages as a business-oriented operating system. However, to attract more businesses we must improve Haiku further, and hence we will delay marketing efforts for businesses until Haiku reaches a stable and mature stage. 

### Partnerships with other Haiku-related sources
If you haven't heard of them already, BeSly and Haikunauts are the two major third party Haiku sources. BeSly is a wiki with tips and tricks on using Haiku, whilst Haikunauts is a blog aimed at promoting Haiku as an OS. As mentioned above, we will work in partnership with these two sources and see what we can do together to help spread the word about Haiku.

Additionally, we are also eager to work with any smaller blogs, websites or channels. If you run a blog, website, channel, or any other source, please contact me via the forums and let me know!

### General

> I am thinking of running an advertisement in _Read the Doc_s, a documentation website that serves a wide array of open-source projects. _Read the Docs_ allocates a portion of their advertising inventory to FOSS projects such as ourselves, hence this is a good opportunity to promote Haiku. What will go on the advertisement has not been decided as of now.

Reviewer: ```suggestion
I am thinking of running an advertisement in _Read the Docs, a documentation website that serves a wide array of open-source projects. _Read the Docs_ allocates a portion of their advertising inventory to FOSS projects such as ourselves, hence this is a good opportunity to promote Haiku. What will go on the advertisement has not been decided as of now.
```

Here, too, a link to the "Read the Docs" website would not be amiss.

scott_puopolo18 is working on a set of interview questions for our developers to give the public a glimpse into the world of Haiku development, as well as shining a light on the challenges and opportunities in an open-source project. The predetermined set of questions means that we don't have to write new questions for each developer. To save time, we can simply send them off and the developer can write their response.

> We are also planning to distribute more content via our often under-utilised social media platforms, as well as update them with current information (i.e. adding profile page art, applying for verification, updating or adding links etc.). Short-form posts on social media are what immediately comes to mind, although we are considering live-streams or videos if there is someone willing to help out.

Reviewer 2: One thing to start with in the video side would be to collect and link the videos of all talks at various conferences (FOSDEM, RMLL, Capitole du Libre, BeGeistert, ...). Most were recorded but I think there is no easy central place to find them.

Haikunauts also has several other ideas for promotional events, but we will share these with you later as we are still working out the details of these events.

### In closing
We are super excited to see how we can promote Haiku and help the project gain new members and volunteers. **[Remember, if you’d like to join, don’t hesitate and send me a private message on the forums!](https://discuss.haiku-os.org/u/jt15s)**

I’d also like to take this opportunity to wish everyone a very Happy New Year - here’s to a more fruitful, happy and lucky new year!

> Finally, I'd like to thank pulkomany and waddlesplash who reviewed this post on Github and gave me plenty of constructive feedback on how I could improve this.

Reviewer 2: PulkoMandy (you're missing a d and the uppercase letters are important :) )

Author: Apologies - I must have been typing this when I was tired :)
